### DISCUSSION TITLE
Grafana and cassandra Error updating options: Unexpected end of JSON input

### DISCUSSION THREAD

**Discussion by bburcuymn** [HadesArchitect/GrafanaCassandraDatasource#184] [General]
I have an issue. I want to visualize my data in Cassandra with Grafana. I made the connection and created a dashboard. I created 2 variables in this dashboard. These are devices and entity. I wrote the query for both. I want the entity query to run according to the selected device data and to receive the appropriate entity data. device is arriving correctly, but my entity data is constantly Templating [entity]. I am getting Error updating options: Unexpected end of JSON input error. When I run the query in the Cassandra terminal, I get the correct results. I have no problems with the query. Whatever the problem may be, I would be very happy if you could help me. Below is the query I wrote for variables:

devices: SELECT table_name FROM system_schema.tables WHERE keyspace_name = 'orion_wins';

entity: SELECT entity_id FROM orion_wins.$devices WHERE entity_type='iot';

grafana log error: level=error msg="Failed to get variables" Message="repo.Select: select query processing: line 1:34 mismatched character 'd' expecting '$'" 

When I run the query in the Cassandra terminal, I get the correct results. I have no problems with the query.

**Comment by unflag** [comment]
Hi @bburcuymn. Currently it is not possible to create dependency between variables. In other words, one variable cannot use another as a part of its query.

**Reply by bburcuymn** [reply]
Is there another way I can use for this?

**Reply by HadesArchitect** [reply]
@unflag do you think it will be hard to implement? Afaiu must be doable, and I have some free time. That's already second question about it, looks like a missing feature, right?

**Reply by unflag** [reply]
> looks like a missing feature, right?

It certainly is!

> do you think it will be hard to implement?

Can't say for sure. Currently each variable is evaluated independently in its own call from frontend to backend. What way grafana provides to create dependency relations between them? That's the question. There should be a way to create dependency graph and sequentially evaluate all variables using it.
